src/igvfd/types/file_set.py

```diff
@@ -118,6 +118,94 @@ class CuratedSet(FileSet):
     schema = load_schema('igvfd:schemas/curated_set.json')
     embedded_with_frame = FileSet.embedded_with_frame
 
+    @calculated_property(
+        schema={
+            'title': 'Assembly',
+            'description': 'The genome assembly to which the referencing files in the file set are utilizing (e.g., GRCh38).',
+            'type': 'array',
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'notSubmittable': True,
+        }
+    )
+    def assembly(self, request, files):
+        assembly_values = set()
+        if files is not None:
+            for current_file_path in files:
+                file_object = request.embed(current_file_path, '@@object?skip_calculated=true')
+                if file_object.get('assembly'):
+                    assembly_values.add(file_object.get('assembly'))
+        assembly_value_list = list(assembly_values)
+        assembly_value_list.sort()
+        if len(assembly_value_list) == 0:
+            assembly_value_list = None
+        return assembly_value_list
```

Comment:
if files is not None:
            assembly_values = set()
            for current_file_path in files:
                file_object = request.embed(current_file_path, '@@object?skip_calculated=true')
                if file_object.get('assembly'):
                    assembly_values.add(file_object.get('assembly'))
        if assembly_values:
             return list(assembly_values)

Reply:
Done, thank you